include/xtensor/xchunked_array.hpp

```diff
@@ -0,0 +1,97 @@
+            auto indexes_of_chunk(std::get<0>(chunk_indexes));
+            auto indexes_in_chunk(std::get<1>(chunk_indexes));
+            chunk_type chunk = m_chunks.element(indexes_of_chunk.cbegin(), indexes_of_chunk.cend());
+            const_reference val = chunk.element(indexes_in_chunk.cbegin(), indexes_in_chunk.cend());
+            int di = 0;
+            for (auto index_of_chunk: indexes_of_chunk)
+            {
+                auto index_in_chunk = indexes_in_chunk[di];
+                std::cout << "Dimension " << di << ": chunk " << index_of_chunk << " at " << index_in_chunk << std::endl;
+                di++;
+            }
+            return val;
+        }
+
+
+        xchunked_array(std::vector<size_t> shape, std::vector<size_t> chunks):
+            m_shape(shape),
+            m_chunk_shape(chunks)
+        {
+            std::vector<size_t> shape_chunk(shape.size());
+            size_t di = 0;
+            std::cout << "Creating chunked array with shape";
+            for (auto s: shape)
+            {
+                std::cout << " " << s;
+                size_t chunk_nb = s / chunks[di];
+                if (s % chunks[di] > 0)
+                    chunk_nb += 1;  // edge chunk
+                shape_chunk[di] = chunk_nb;
+                di++;
+            }
+            std::cout << ", chunk shape";
+            for (auto s: chunks)
+                std::cout << " " << s;
+            std::cout << std::endl;
+            m_chunks.resize(shape_chunk);
+        }
+
+    private:
+
+        template <class Dim, class Idx>
+        std::tuple<size_t, size_t> get_chunk_indexes_in_dimension(Dim dim, Idx idx) const
+        {
+            std::cout << " " << dim;
+            size_t index_of_chunk = idx / m_chunk_shape[dim];
+            size_t index_in_chunk = idx - index_of_chunk * m_chunk_shape[dim];
+            return std::make_tuple(index_of_chunk, index_in_chunk);
+        }
+
+        template <size_t... dims, class... Idxs>
+        std::array<std::tuple<size_t, size_t>, sizeof...(Idxs)>
+        get_chunk_indexes(std::index_sequence<dims...>, Idxs... idxs) const
+        {
+            std::array<std::tuple<size_t, size_t>, sizeof...(Idxs)> chunk_indexes = {{get_chunk_indexes_in_dimension(dims, idxs)...}};
+            return chunk_indexes;
+        }
+
+        template <class T, size_t N, size_t... Is>
+        std::tuple<std::array<size_t, N>, std::array<size_t, N>> unpack_impl(std::array<T, N> &arr, std::index_sequence<Is...>) const
+        {
+            std::array<size_t, N> arr0 = {{std::get<0>(arr[Is])...}};
+            std::array<size_t, N> arr1 = {{std::get<1>(arr[Is])...}};
+            return std::make_tuple(arr0, arr1);
+        }
+
+        template <class T, std::size_t N>
+        std::tuple<std::array<size_t, N>, std::array<size_t, N>> unpack(std::array<T, N> &arr) const
+        {
+            return unpack_impl(arr, std::make_index_sequence<N>());
+        }
```

Comment:
I think you can make a simpler implementation like
```cpp
std::transform(arr.begin(), arr.end(), arr0.begin() [](const auto& v) { return std::get<0>(v); });
std::transform(arr.begin(), arr.end(), arr1.begin() [](const auto& v) { return std::get<1>(v); });
```

Or if you want to avoid traversing the array twice:

```cpp
for(size_t i = 0; i < N; ++i)
{
    arr0[i] = std::get<0>(arr[i]);
    arr1[i] = std::get<1>(arr[i]);
}
```

Reply:
:+1: